### PR TITLE
Added an index node in managers configuration

### LINKED ISSUE #661: Merge connections node to the manager in configuration
This is a proposal to remove connections node in configuration and define a connection with current spec directly in the manager. Since this configuration was introduced it add's only duplication and additional effort to define the index. I have never been using the single connection in different managers. And if in any case, it would happen user can still use parameters for index name or other params.

@ongr-io/team wdyt?

### PR DESCRIPTION
**Pull request by einorler**
Closes #661

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -89,7 +89,6 @@ private function getConnectionsNode()
         $node = $builder->root('connections');
 
         $node
-            ->isRequired()
             ->requiresAtLeastOneElement()
```
Comment (saimaz) [inline DependencyInjection/Configuration.php]: Change to `defautl(null)`

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -158,9 +157,53 @@ private function getManagersNode()
             ->prototype('array')
                 ->children()
                     ->scalarNode('connection')
-                        ->isRequired()
                         ->info('Sets connection for manager.')
                     ->end()
+                    ->arrayNode('index')
+                        ->children()
+                            ->scalarNode('name')
```
Comment (saimaz) [inline DependencyInjection/Configuration.php]: change to `index_name`

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -158,9 +157,53 @@ private function getManagersNode()
             ->prototype('array')
                 ->children()
                     ->scalarNode('connection')
-                        ->isRequired()
                         ->info('Sets connection for manager.')
                     ->end()
+                    ->arrayNode('index')
```
Comment (saimaz) [inline DependencyInjection/Configuration.php]: Reuse `getConnectionsNode` by introducing node name parameter.

**Comment by einorler** [comment]
Changed `index` node `name` to `index_name`, refactored `process` method of the compiler pass, added a default value to the `connections` node, updated a test

**Comment by saimaz** [comment]
@einorler can you update the docs with changes of this PR?

**Comment by einorler** [comment]
Sure

**Pull request merged**